packages/indexing/src/database-index-api.ts

```diff
@@ -115,49 +144,81 @@ export abstract class DatabaseIndexApi<DateType = Date | number> {
    * @param models
    */
   async indexModels(models: Array<IndexModelArgs>): Promise<void> {
-    await this.indexModelsInDatabase(models)
-    for (const modelArgs of models) {
-      await this.assertNoOngoingSyncForModel(modelArgs.model)
-      const foundModelToIndex = this.indexedModels.find((indexedModel) =>
-        indexedModel.streamID.equals(modelArgs.model)
-      )
-      if (!foundModelToIndex) {
-        this.indexedModels.push({
-          streamID: modelArgs.model,
-          indices: modelArgs.indices,
-        })
+    // Updated indexed models record
+    const indexedModels: Record<string, ModelData> = cloneDeep(this.indexedModelsRecord)
+    // Models to index with merged indices
+    const modelsToIndex: Array<IndexModelArgs> = []
+    // Record of indices to create
+    const createIndices: CreateIndicesRecord = {}
+    // Checks for models in sync
+    const checkSyncPromises: Array<Promise<void>> = []
+
+    for (const modelData of models) {
+      // Check model is not currently going through historical sync
```

Comment:
what will we need to do when it does?  is there a todo here ?

Reply:
If there is any sync in progress, the indices creation will fail, that is covered by https://github.com/ceramicnetwork/js-ceramic/pull/3067/files#diff-bfb31d44a4dee8c373a193980ab3484df1782c7d3ab07c5c5b54e47886ef7aeaR192 that ensures the attempt to sync

Reply:
ok should we add a comment about what happens whether the indexes are created later or they need to be manually created, like will this be ok or does someone need to do something?

Reply:
It should throw an error if the sync is in progress, so the whole operation will fail and the person should try again later or stop the sync before trying again.